apps/antalmanac/src/components/dialogs/CopySchedule.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect, useCallback } from 'react';
+import {
+    Box,
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    MenuItem,
+    Select,
+    type DialogProps,
+} from '@mui/material';
+import AppStore from '$stores/AppStore';
+import { copySchedule } from '$actions/AppStoreActions';
+
+interface CopyScheduleDialogProps extends DialogProps {
+    index: number;
+}
+
+function CopyScheduleDialog(props: CopyScheduleDialogProps) {
+    const { index, onClose } = props;
```

Comment:
Separately destructure onClose please; (Reference `DeleteSchedule.tsx` or ask Brian if curious why)

```suggestion
    const { index } = props;
    
     /**
     * This is destructured separately for memoization.
     */
    const { onClose } = props;
```